ads-cert-examples/scala/src/main/scala/Main.scala

```diff
@@ -0,0 +1,87 @@
+import java.io.{FileInputStream, InputStreamReader}
+import java.nio.charset.StandardCharsets
+import java.security.spec.X509EncodedKeySpec
+import java.security._
+
+import org.apache.commons.codec.binary.Base64
+
+import org.bouncycastle.asn1.ASN1ObjectIdentifier
+import org.bouncycastle.asn1.x9.ECNamedCurveTable
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter
+import org.bouncycastle.openssl.{PEMKeyPair, PEMParser}
+
+import scala.util.{Failure, Success, Try}
+
+
+object Main {
+
+  private var privateKey: PrivateKey = _
+  private var publicKey: PublicKey = _
+
+  Security.addProvider(new BouncyCastleProvider())
+
+  def main(args: Array[String]): Unit = {
+    loadKeys(args)
+    val message = Stream(
+      "2824f5d3-a2dc-49ec-8358-dbe92bd01ec2", // source.tid
+      "a:180720yz:1807241326", // ad.pcv
+      "newsite.com", // site.domain
+      "", // app.bundle
+      "", // user.consent
+      "vd", // placement
+      "192.168.1.1", // device.ip
+      "", // device.ipv6
+      "", // device.ifa
+      "", // device.ua
+      "" // ad.video player size?
+    ).map(Option(_).getOrElse("")).toList.mkString(":")
+    for {
+      signature <- sign(Signature.getInstance("SHA256withECDSA"), message)
+    } yield {
+      printf("publisher signature: %s\n", Base64.encodeBase64String(signature))
+      Try {
+        verify(Signature.getInstance("SHA256withECDSA"), message, signature)
+      } match {
+        case Success(b) => b.getOrElse(false) match {
+          case true => println("yay it worked!")
+          case false => println("it didn't work :(")
+        }
+        case Failure(e) => throw e
+      }
+    }
+  }
+
+  private def loadKeys(args: Array[String]): Try[_] = {
+    if (args.length < 2) {
+      Failure(new IllegalArgumentException("invalid argument length"))
+    }
+    val privateParser = new PEMParser(new InputStreamReader(new FileInputStream(args(0))))
+    val oid = privateParser.readObject.asInstanceOf[ASN1ObjectIdentifier]
```

Comment:
Unchecked casts through `asInstanceOf` is considered bad practice.  Using a pattern match to explicitly handle the failure would be more common.  Since we're already using `Try` then it might make sense to generate a `Try[ASN1ObjectIdentifier]` and then flatmap the result with downstream processing.